manifests/dnsserver.pp

```diff
@@ -4,7 +4,7 @@
   $ip,
   $domain = undef,
 ) {
-  if !is_ip_address($ip) { fail("Expect IP address for ip, got ${ip}") }
+  validate_re($ip, '^\d+.\d+.\d+.\d+(#\d+)?$')
```

Comment:
we can provide more than just an ip here, dnsmasq allows the specification of a port by x.x.x.x#port. see http://linux.die.net/man/8/dnsmasq